app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt

```diff
@@ -74,46 +75,186 @@ class RefreshController @Inject constructor(
         stopped: (() -> Unit)? = null,
     ): Pair<Set<Thread>?, Throwable?> {
 
-        suspend fun refreshWithRunCatching(job: Job, isFirstTime: Boolean = true): Pair<Set<Thread>?, Throwable?> = runCatching {
-            withContext(Dispatchers.IO + job) {
-                if (isFirstTime) {
-                    started?.invoke()
-                } else {
-                    delay(Utils.DELAY_BEFORE_FETCHING_ACTIVITIES_AGAIN)
-                    ensureActive()
+        SentryLog.i("API", "Refresh threads with mode: $refreshMode | (${folder.name})")
+
+        endOfMessagesReached = false
+
+        refreshThreadsJob?.cancel()
+        refreshThreadsJob = Job()
+
+        return refreshWithRunCatching(refreshThreadsJob!!, refreshMode, mailbox, folder, okHttpClient, realm, started, stopped)
+            .also {
+                it.first?.let {
+                    stopped?.invoke()
+                    SentryLog.d("API", "End of refreshing threads with mode: $refreshMode | (${folder.name})")
                 }
-                realm.handleRefreshMode(refreshMode, scope = this, mailbox, folder, okHttpClient) to null
             }
-        }.getOrElse {
-            // If fetching the activities failed because of a not found Message, we should pause briefly
-            // before trying again to retrieve activities, to ensure that the API is up-to-date.
-            if (isFirstTime && it is MessageNotFoundException) {
-                Sentry.withScope { scope ->
-                    scope.level = SentryLevel.WARNING
-                    scope.setTag("isFirstTime", "true")
-                    scope.setExtra("folderCursor", "${folder.cursor}")
-                    scope.setExtra("folderName", folder.name)
-                    Sentry.captureException(it)
-                }
-                refreshWithRunCatching(job, isFirstTime = false)
-            } else {
-                handleAllExceptions(it, stopped, folder)
+    }
+
+    private suspend fun refreshWithRunCatching(
+        job: Job,
+        refreshMode: RefreshMode,
+        mailbox: Mailbox,
+        folder: Folder,
+        okHttpClient: OkHttpClient?,
+        realm: Realm,
+        started: (() -> Unit)?,
+        stopped: (() -> Unit)?,
+    ): Pair<Set<Thread>?, Throwable?> = runCatching {
+        withContext(Dispatchers.IO + job) {
+            started?.invoke()
+            realm.handleRefreshMode(
+                refreshMode = refreshMode,
+                scope = this,
+                mailbox = mailbox,
+                folder = folder,
+                okHttpClient = okHttpClient,
+            ) to null
```

Comment:
```suggestion
            return@withContext realm.handleRefreshMode(refreshMode, mailbox, folder, okHttpClient, scope = this) to null
```
Or maybe even this : 
```suggestion
            val threads =  realm.handleRefreshMode(refreshMode, mailbox, folder, okHttpClient, scope = this)
            
            return@withContext threads to null
```